core/src/main/java/nl/nn/adapterframework/extensions/esb/EsbJmsListener.java

```diff
@@ -136,34 +136,37 @@ public Map<String, Object> extractMessageProperties(Message rawMessage) throws L
 			}
 		}
 
-		try {
-			String soapMessage;
-			if (rawMessage instanceof TextMessage) {
-				TextMessage textMessage = (TextMessage) rawMessage;
-				soapMessage = textMessage.getText();
-			} else if (rawMessage instanceof BytesMessage) {
-				BytesMessage bytesMessage = (BytesMessage) rawMessage;
-				InputStream input = new BytesMessageInputStream(bytesMessage);
-				soapMessage = StreamUtil.streamToString(input);
-			} else {
-				soapMessage = null;
-			}
+		if (!getxPathLogMap().isEmpty()) {
+			try {
+				String soapMessage;
+				if (rawMessage instanceof TextMessage) {
+					TextMessage textMessage = (TextMessage) rawMessage;
+					soapMessage = textMessage.getText();
+				} else if (rawMessage instanceof BytesMessage) {
+					BytesMessage bytesMessage = (BytesMessage) rawMessage;
+					InputStream input = new BytesMessageInputStream(bytesMessage);
+					soapMessage = StreamUtil.streamToString(input);
+					bytesMessage.reset();
+				} else {
+					soapMessage = null;
```

Comment:
Just wondering, is this explicit `null` assignment needed or is it just a code style preference?

Reply:
I thought that a `String` is by default `null` when it's not initialized right?

Reply:
It's a local variable, not a field in the class. So when not initialized it has no value. You cannot access it unless it's explicitly assigned -- the compiler will give an error.

Reply:
I understand, although I think it's a bit silly that local variables behave differently.

Thanks for the clarification :)

Reply:
Actually I'm glad they behave the way they do, as it allows to catch accidentally unitialized variables.

Reply:
> Actually I'm glad they behave the way they do, as it allows to catch accidentally unitialized variables.

That's a good argument.

For me it's the inconsistency, apparently only null is assigned to class member variables.. why wouldn't the same concept apply here?

Reply:
Yes, I agree.